src/index.js

```diff
@@ -306,18 +306,19 @@ function shimBelongsToMany(target) {
   shimmer.wrap(target, 'get', original => {
     return function bathedGetHasMany(instances, options = {}) {
       let isCount = false;
-      assert(this.paired, '.paired missing on belongsToMany association. You need to set up both sides of the association');
 
       if (options.include || options.transaction || activeClsTransaction() || !options[EXPECTED_OPTIONS_KEY]) {
         return original.apply(this, arguments);
       }
 
+      const manyFromSourceAs = this.paired ? this.paired.manyFromSource.as : this.manyFromSource.as;
```

Comment:
Why do you expect this to work?
`manyFromSource` expresses the relationship from the model to the join table, and we specifically need it for the other side to load the other side, unless both sides use the same foreign keys etc it wouldn't work.

Reply:
You'd probably need something like `manyFromTarget` but i'm not sure that Sequelize exposes that.

Reply:
I just introduced this variable, the value is used later in your code.  The `paired` is null when the relationship/association is not declared both way (in my case user has many roles, but role is oblivious to existence of the user: `User -> Role` through `UserRole`).

Reply:
I'm talking about `manyFromSource` in Sequelize, not your variable, it's evident from the code :)
You can try to make it work, but the changes you made here would never work as `this.manyFromSource` gives you the wrong side of the association, as i wrote :)

Reply:
Now I have found out that that issue maybe caused by my "fix" ... When dataloader is not involved, the correct data is returned and the `JOIN` statement condition is correct. I'll dive into it.

Reply:
Well yes that's what i pointed out, that the code you introduced here would never work.